src/client/api/event.c

```diff
@@ -1261,18 +1264,24 @@ daos_event_priv_wait()
 		/** progress succeeded, loop can exit if event completed */
 		if (rc == 0) {
 			rc = ev_thpriv.ev_error;
-			if (rc)
+			if (rc) {
+				rc2 = daos_event_priv_reset();
```

Comment:
why reset it here?
the "ev_thpriv.ev_error != 0" just means the async operation failed right? reset will lost the fail code, no?

Reply:
the error itself is saved first returned from the function itself, so it won't get lost?
the purpose of this is a strange bug i saw in :
https://github.com/daos-stack/daos/pull/7177
where a DAOS operation was failing (as expected) but the error code was getting saved and returned in a subsequent operation (that has passed, but still reported the error at the API level from the previous operation).

Reply:
ok, maybe reset ev_thpriv.ev_error = 0 is enough, but daos_event_priv_reset() seems no bad effect, just seems a little overkill.

Reply:
i see. i can try that in the other PR after this lands and i merge there.